src/components/organisms/AdminVenueView/components/EventsView/CsvImportModal.tsx

```diff
@@ -0,0 +1,219 @@
+import React, { useCallback, useState } from "react";
+import { Modal } from "react-bootstrap";
+import { useAsyncFn } from "react-use";
+import classNames from "classnames";
+import { ParseError, ParseMeta } from "papaparse";
+
+import { importEventsBatch, ImportEventsBatchResult } from "api/events";
+
+import { parseCsv, ParseData, transformImportToEvents } from "utils/csv";
+
+import { ButtonNG } from "components/atoms/ButtonNG/ButtonNG";
+import {
+  FileButton,
+  FileButtonOnChangeData,
+} from "components/atoms/FileButton/FileButton";
+
+import "./CsvImportModal.scss";
+
+export interface CsvImportModalProps {
+  onHide: () => void;
+  show: boolean;
+}
+
+export const CsvImportModal: React.FC<CsvImportModalProps> = ({
+  onHide,
+  show,
+}) => {
+  const [parsedData, setParsedData] = useState<ParseData[] | undefined>();
+  const [parsedMeta, setParsedMeta] = useState<ParseMeta | undefined>();
+  const [parsedErrors, setParsedErrors] = useState<ParseError[] | undefined>();
+  const [isParsing, setIsParsing] = useState<boolean>(false);
+  const [importResult, setImportResult] = useState<ImportEventsBatchResult>();
+
+  const clearParsed = useCallback(() => {
+    setParsedData(undefined);
+    setParsedMeta(undefined);
+    setParsedErrors(undefined);
+    setImportResult(undefined);
+  }, []);
+
+  const downloadExample = useCallback(() => {
+    const link = document.createElement("a");
+    link.download = "example.csv";
+    link.href = "/csv/import-events-example.csv";
+    link.click();
+    link.remove();
+  }, []);
+
+  const selectFile = useCallback(
```

Comment:
You can use `useAsyncFn` here and use it's loading state instead of creating a new state for it.

Reply:
done @ 18f28f5c4b763ca615464e201bea5daab92ffeae

too bad `useAsyncFn` adds artificial limits on the type of rejection reasons